### PR TITLE
Add boskos related helpers for cluster management lib

### PR DESCRIPTION
**Pull request by chaodaiG**
Creating this helper for defining some constants, as well as certain common contracts, i.e. using `JOB_NAME` env var as host name for acquiring Boskos resources

Part of: https://github.com/knative/test-infra/issues/1186

/cc @adrcunha

### CODE REVIEW

**Comment by adrcunha** [comment]
Tests, pretty please? :)

**Comment by chaodaiG** [comment]
Do you mean testing boskos helpers? I have thought about it, but it doesn't really carry any custom logic other than wrapping around k8s/boskos, I can add tests for the constants though if desired.

**Comment by adrcunha** [comment]
> Do you mean testing boskos helpers? I have thought about it, but it doesn't really carry any custom logic other than wrapping around k8s/boskos, I can add tests for the constants though if desired.

I would be more comfortable with tests for `AcquireGKEProject` and `ReleaseGKEProject`. And not only for themselves, but the ability to mock the boskos library will probably be very useful in more complex tests later (e.g., cleanup functions).

**Comment by chaodaiG** [comment]
@adrcunha , done adding unit tests

File: testutils/clustermanager/boskos/boskos.go [hunk 1]
```diff
@@ -16,14 +16,60 @@ limitations under the License.
...
+
+func newClient(host *string) *boskosclient.Client {
+	if nil == host {
+		host = &defaultHost
+	}
+	return boskosclient.NewClient(*host, boskosURI)
+}
+
+// AcquireGKEProject acquires GKE Boskos Project with "free" state, and not
+// owned by anyone, sets its state to "busy" and assign it an owmner of *host,
```
Comment (adrcunha) [inline testutils/clustermanager/boskos/boskos.go]:
```suggestion
// owned by anyone, sets its state to "busy" and assign it an owner of *host,
```

File: testutils/clustermanager/boskos/boskos_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		expErr    bool
+	}{
+		{true, true},
+		{false, false},
+	}
+	oldBoskosURI := boskosURI
+	defer func() {
+		boskosURI = oldBoskosURI
+	}()
+	for _, data := range datas {
```
Comment (adrcunha) [inline testutils/clustermanager/boskos/boskos_test.go]: The most important part in this test should be testing the contract about the host parameter, as you described in the comment.

File: testutils/clustermanager/boskos/boskos_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			if data.serverErr {
+				http.Error(w, "", http.StatusBadRequest)
+			} else {
+				fmt.Fprint(w, "")
+			}
+		})
+		defer ts.Close()
+		boskosURI = ts.URL
+		err := ReleaseGKEProject(&fakeHost, "foobar")
+		if data.expErr && (nil == err) {
```
Comment (adrcunha) [inline testutils/clustermanager/boskos/boskos_test.go]: As you described, this function is too powerful. The most important part of this test is to ensure that the proper parameters (host, name) are passed to the boskos library.

Comment (chaodaiG) [reply testutils/clustermanager/boskos/boskos_test.go]: done

**Comment by adrcunha** [comment]
Thanks for the test coverage.

/lgtm
/approve

**Pull request merged**